File: kotlin/src/test/kotlin/com/joshy/hywrapper/util/UuidUtilsTest.kt
package com.joshy.hywrapper.util

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class UuidUtilsTest {
    private val dashed = "550e8400-e29b-41d4-a716-446655440000"
    private val undashed = "550e8400e29b41d4a716446655440000"

    @Test
    fun testUndash() {
        assertEquals(undashed, UuidUtils.undash(dashed))
        assertEquals(undashed, UuidUtils.undash(undashed))
    }

    @Test
    fun testDash() {
        assertEquals(dashed, UuidUtils.dash(undashed))
        assertEquals(dashed, UuidUtils.dash(dashed))
    }

    @Test
    fun testIsValid() {
        assertTrue(UuidUtils.isValid(dashed))
        assertTrue(UuidUtils.isValid(undashed))
        assertFalse(UuidUtils.isValid("invalid-uuid"))
        assertFalse(UuidUtils.isValid("550e8400-e29b-41d4-a716-44665544000g")) // invalid hex
    }
}

File: typescript/tests/uuid_utils.test.ts
import { UuidUtils } from "../src/utils/uuid_utils";

describe("UuidUtils", () => {
    const dashed = "550e8400-e29b-41d4-a716-446655440000";
    const undashed = "550e8400e29b41d4a716446655440000";

    test("undash should remove dashes and convert to lowercase", () => {
        expect(UuidUtils.undash(dashed)).toBe(undashed);
        expect(UuidUtils.undash(undashed)).toBe(undashed);
    });

    test("dash should add dashes to an undashed UUID", () => {
        expect(UuidUtils.dash(undashed)).toBe(dashed);
        expect(UuidUtils.dash(dashed)).toBe(dashed);
    });

    test("isValid should validate UUIDs correctly", () => {
        expect(UuidUtils.isValid(dashed)).toBe(true);
        expect(UuidUtils.isValid(undashed)).toBe(true);
        expect(UuidUtils.isValid("invalid-uuid")).toBe(false);
        expect(UuidUtils.isValid("550e8400-e29b-41d4-a716-44665544000g")).toBe(false);
    });
});
